### ISSUE TITLE
False warning `script file doesn't have execute permission`

### ISSUE THREAD

**Issue by mvasin** [bbatsov/rubocop#4526]
## Expected behavior

A script file with shebang should not rise "Script file doesn't have execute permission" if it does have `x` bits (after running `chmod +x filename`).

## Actual behavior

The file is executable
```
$ ls -la | grep app
-rwxr-xr-x   1 mike  staff   81 Jun 17 15:05 app.rb
```
but the warning is raised:

```
$ rubocop
Inspecting 4 files
.W..

Offenses:

app.rb:1:1: W: Script file app.rb doesn't have execute permission.
#!/usr/bin/env ruby
^^^^^^^^^^^^^^^^^^^

4 files inspected, 1 offense detected
```

I double check it the way rubocop checks it, and it's executable for sure:

```
$ irb
irb(main):001:0> File.stat('./app.rb').executable?
=> true
```

## Steps to reproduce the problem

Create a file with shebang, run `chmod +x filename` and run `rubocop`.

## RuboCop version
```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by pocke** [comment]
Could you try to run rubocop with `--cache false` option?
Maybe this problem is same as https://github.com/bbatsov/rubocop/issues/4469

**Comment by mvasin** [comment]
@pocke Thanks, it worked. I'll close this issue as duplicate.

**Comment by CreeGunning** [comment]
@pocke @mvasin Hey guys, I'm, still having problems with this. Simply trying to pass my Linting and get CI CD working for a college project, however I keep getting the below error.

Offenses:

bin/bundle:1:1: W: Lint/ScriptPermission: Script file bundle doesn't have execute permission.
#!/usr/bin/env ruby
^^^^^^^^^^^^^^^^^^^
bin/importmap:1:1: W: Lint/ScriptPermission: Script file importmap doesn't have execute permission.
#!/usr/bin/env ruby
^^^^^^^^^^^^^^^^^^^
bin/rails:1:1: W: Lint/ScriptPermission: Script file rails doesn't have execute permission.
#!/usr/bin/env ruby.exe
^^^^^^^^^^^^^^^^^^^^^^^
bin/rake:1:1: W: Lint/ScriptPermission: Script file rake doesn't have execute permission.
#!/usr/bin/env ruby.exe
^^^^^^^^^^^^^^^^^^^^^^^
bin/setup:1:1: W: Lint/ScriptPermission: Script file setup doesn't have execute permission.
#!/usr/bin/env ruby.exe
^^^^^^^^^^^^^^^^^^^^^^^

96 files inspected, 5 offenses detected
Error: Process completed with exit code 1.

I'm not too sure why this is happening. The fixes about don't seem to solve my problem. Any thoughts guys?

**Comment by ahmedeid6842** [comment]
@CreeGunning :wave: 
Did you find a solution for you problem because I'm facing the same problem :question:

**Comment by GeekyHacks** [comment]
Same here getting the 
Offenses:

main_school:1:1: W: Lint/ScriptPermission: Script file main_school doesn't have execute permission.
#!/usr/bin/ruby
^^^^^^^^^^^^^^^

**Comment by GeekyHacks** [comment]
you can add `# rubocop:disable BlockComments, AsciiComments` at the beginning of the file you have `#!/usr/bin/ruby` and it should work fine and rupucop wont detect any comments on that file.

you can refer to the [StackOverflow](https://stackoverflow.com/questions/26342570/rubocop-line-length-how-to-ignore-lines-with-comments)
